Dockerfile.builder

```diff
@@ -0,0 +1,52 @@
+#
```

Comment:
Can you leave a description where is this image used?

Reply:
Added comment:
```
#
# Cilium build-time dependencies.
# Image created from this file is used to build Cilium.
#
```

Note that while I removed the most obvious Envoy dependencies from here, I'm not sure if all that remains is really needed to build Cilium. Maybe you can have look?

Reply:
@aanm ^^^^

Reply:
I know it's used to build Cilium :-) I'm just not sure if this one is the cilium/builder or the cilium/runtime